sstables/sortedmap_key_index.go

```diff
@@ -0,0 +1,128 @@
+package sstables
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+
+	"golang.org/x/exp/slices"
+
+	rProto "github.com/thomasjungblut/go-sstables/recordio/proto"
+	"github.com/thomasjungblut/go-sstables/skiplist"
+	"github.com/thomasjungblut/go-sstables/sstables/proto"
+)
+
+// SortedMapIndex is keeping the entire index as a map and the ordered keys as a slice in memory
+// Use binary search to iter over the index.
+type SortedMapIndex struct {
```

Comment:
I wonder, could we have a generic type parameter to also make the [20]byte become configurable?

> type SortedMapIndex[T comparable] struct {

or so?

Reply:
I have done that. But how can i implement a ByteKeyMapper[[20]byte] with my own MapBytes function ? I need to find how to do that to complete my PR

Reply:
done

Reply:
okay wow! thank you :) I need a bit of time to go through all of this

Reply:
I don't know whether we can split this somehow in meaningful pieces? the benchmark can be separated, it also seems the index itself can go separately and there is a new "version" of how to write the sstable